projects/storefrontlib/src/cms-components/product/product-interests/stock-notification/stock-notification.component.spec.ts

```diff
@@ -183,129 +154,108 @@ describe('StockNotificationComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should not load stock notificaiton data, when customer is not login', () => {
-    authService.getUserToken.and.returnValue(of({}));
+  it('should show disabled button and link for login custom without channel set', () => {
+    userService.getNotificationPreferences.and.returnValue(of([]));
     fixture.detectChanges();
-    component.logged$.subscribe(x => expect(x).toBeFalsy());
-  });
 
-  it('should load stock notification data , when customer is login', () => {
-    authService.getUserToken.and.returnValue(of(userToken));
-    fixture.detectChanges();
-    component.logged$.subscribe(x => expect(x).toBeTruthy());
+    expect(
+      el.query(By.css('[data-test]="stocknotify-link"')).nativeElement
```

Comment:
the description should change to 'show disabled button and setting link for login customer without setting notification channel'
the element naming should be meaningful, don't alway use stocknotify as prefix